framework/src/base/FEProblemBase.C

```diff
@@ -4428,12 +4431,14 @@ FEProblemBase::adaptMesh()
     else
     {
       _console << "Mesh unchanged, skipping remaining steps..." << std::endl;
-      return;
+      break;
```

Comment:
Well unfortunately I'm also to thank for putting it in there in the first place!